### PR TITLE
Fix unused xml_id for files

### LINKED ISSUE #807: XML ID of Task Files not used for export
When exporting a task with regular task files, we do not use the XML ID for the task files:

https://github.com/openHPI/codeharbor/blob/75a0b1e677d20b07e028de444f14042f38a7a0b1/app/services/proforma_service/convert_task_to_proforma_task.rb#L71-L82

Instead, we just use the database ID of the file:

https://github.com/openHPI/codeharbor/blob/75a0b1e677d20b07e028de444f14042f38a7a0b1/app/services/proforma_service/convert_task_to_proforma_task.rb#L73

This causes issues when providing additional metadata, such as a desired file role, as they are not linked together correctly.

### PR DESCRIPTION
**Pull request by kkoehn**
xml_id was added to task_files
xml_id is now required for files (the same as tests and model_solutions already) and will be automatically filled out in the form with a default value

fixes #807 
helps #808

### CODE REVIEW

File: app/models/task_file.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class TaskFile < ApplicationRecord
   has_one_attached :attachment
   validates :name, presence: true
   validates :attachment, presence: true, if: -> { use_attached_file == 'true' }, on: :force_validations
+  validates :xml_id, presence: true
+  validates :xml_id, uniqueness: {scope: %i[fileable_id fileable_type]}
```
Comment (MrSerth) [inline app/models/task_file.rb:10]: At least in my tests, this validation doesn't work as expected. I was able to add two new files with the same XML ID and save those. Only when editing, this validation kicks in and prevents saving the task. Do we skip the validation for new task_files for some reason?

Bildschirm­foto 2022-12-17 um 10 29 32
Bildschirm­foto 2022-12-17 um 10 30 25

File: app/models/task_file.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ class TaskFile < ApplicationRecord
   has_one_attached :attachment
   validates :name, presence: true
   validates :attachment, presence: true, if: -> { use_attached_file == 'true' }, on: :force_validations
+  validates :xml_id, presence: true
```
Comment (MrSerth) [inline app/models/task_file.rb:9]: If we force the XML ID to be present, shouldn't we always display it? And what about (a migration for) existing data? We could clear the test data, but probably should consider this as well.

Comment (MrSerth) [reply app/models/task_file.rb:9]: Indeed, existing exercises break:

Bildschirm­foto 2022-12-17 um 10 38 36

```
ActiveRecord::RecordInvalid - Validation failed: Files xml can't be blank, Tests files xml can't be blank, Model solutions files xml can't be blank
```

File: app/models/task_file.rb (hunk 1 above)
Comment (MrSerth) [reply app/models/task_file.rb:10]: Additionally, the scope is probably not sufficient yet. The following entry is valid according to the validation:

Bildschirm­foto 2022-12-17 um 10 45 20

but generates an invalid XML:

Bildschirm­foto 2022-12-17 um 10 44 45

```
Proforma::PostGenerateValidationError - [#<Nokogiri::XML::SyntaxError: 36:0: ERROR: Element '{urn:proforma:v2.0.1}file': Duplicate key-sequence ['0'] in key identity-constraint '{urn:proforma:v2.0.1}task-fileids'.>]
```

**Review by MrSerth — CHANGES_REQUESTED** [review]
Thanks for working on the XML ID integration for files! Despite finding some issues with the current validation of user data, I was successfully able to verify your fix regarding the original issue #807. 💪 

Please have another look at the issues I found with this PR; we might also need to change the seed data to either accommodate or auto-generate an XML ID if not created yet.

File: app/models/task_file.rb (hunk 1 above)
Comment (kkoehn) [reply app/models/task_file.rb]: I added a custom validation that should cover this

**Review by MrSerth — APPROVED** [review]
Thanks! I tested various scenarios on my system and think this is ready to go now 🏁 💪

**Comment by MrSerth** [comment]
I just rebased this branch (to test with the latest Ruby 3.2) and will merge now.

**Pull request merged**